docs/functionaldatatypes/nonemptylist.md

```diff
@@ -0,0 +1,195 @@
+---
+id: nonemptylist
+title: "NonEmptyList"
+---
+
+`NonEmptyList` represents a `List` that is guaranteed not to be empty.
+
+This is useful to allow more accurate domain modeling in situations where we know a collection is not empty.
+
+For example, we might be handling a collection of errors.
+
+If an error has occurred at all then we know the collection is not empty. However, the Scala compiler does not know that and so we are forced to handle the case where the collection is empty even though we know that cannot occur.
```

Comment:
pedantic: it's a little hand-wavy that Scala compiler does not know that without stating the standard library collections we have in mind (List, Vector, Iterable) though clearly just about everybody would think of those anyhow.

Reply:
Changed to incorporate specific collections.